### PR TITLE
Fetch TVL for all chains Beluga supports

### PR DESCRIPTION
**Pull request by chainvisions**

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/beluga: 
        
 
  
    
    
    Error: Request failed with status code 503
        at createError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/axios/lib/core/createError.js:16:15)
        at settle (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/axios/lib/core/settle.js:17:12)
        at IncomingMessage.handleStreamEnd (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/axios/lib/adapters/http.js:322:11)
        at IncomingMessage.emit (node:events:538:35)
        at endReadableNT (node:internal/streams/readable:1345:12)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      config: {
        transitional: {
          silentJSONParsing: true,
          forcedJSONParsing: true,
          clarifyTimeoutError: false
        },
        adapter: [Function: httpAdapter],
        transformRequest: [ [Function: transformRequest] ],
        transformResponse: [ [Function: transformResponse] ],
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        maxBodyLength: -1,
        validateStatus: [Function: validateStatus],
        headers: {
          Accept: 'application/json, text/plain, */*',
          'User-Agent': 'axios/0.26.1'
        },
        method: 'get',
        url: 'https://api.beluga.fi/tvl',
        data: undefined
      },
      request: <ref *1> ClientRequest {
        _events: [Object: null prototype] {
          abort: [Function (anonymous)],
          aborted: [Function (anonymous)],
          connect: [Function (anonymous)],
          error: [Function (anonymous)],
          socket: [Function (anonymous)],
          timeout: [Function (anonymous)],
          prefinish: [Function: requestOnPrefinish]
        },
        _eventsCount: 7,
        _maxListeners: undefined,
        outputData: [],
        outputSize: 0,
        writable: true,
        destroyed: false,
        _last: true,
        chunkedEncoding: false,
        shouldKeepAlive: false,
        maxRequestsOnConnectionReached: false,
        _defaultKeepAlive: true,
        useChunkedEncodingByDefault: false,
        sendDate: false,
        _removedConnection: false,
        _removedContLen: false,
        _removedTE: false,
        _contentLength: 0,
        _hasBody: true,
        _trailer: '',
        finished: true,
        _headerSent: true,
        _closed: false,
        socket: TLSSocket {
          _tlsOptions: [Object],
          _secureEstablished: true,
          _securePending: false,
          _newSessionPending: false,
          _controlReleased: true,
          secureConnecting: false,
          _SNICallback: null,
          servername: 'api.beluga.fi',
          alpnProtocol: false,
          authorized: true,
          authorizationError: null,
          encrypted: true,
          _events: [Object: null prototype],
          _eventsCount: 11,
          connecting: false,
          _hadError: false,
          _parent: null,
          _host: 'api.beluga.fi',
          _readableState: [ReadableState],
          _maxListeners: undefined,
          _writableState: [WritableState],
          allowHalfOpen: false,
          _sockname: null,
          _pendingData: null,
          _pendingEncoding: '',
          server: undefined,
          _server: null,
          ssl: null,
          _requestCert: true,
          _rejectUnauthorized: true,
          parser: null,
          _httpMessage: [Circular *1],
          write: [Function: writeAfterFIN],
          [Symbol(res)]: [TLSWrap],
          [Symbol(verified)]: true,
          [Symbol(pendingSession)]: null,
          [Symbol(async_id_symbol)]: 114,
          [Symbol(kHandle)]: null,
          [Symbol(kSetNoDelay)]: false,
          [Symbol(lastWriteQueueSize)]: 0,
          [Symbol(timeout)]: null,
          [Symbol(kBuffer)]: null,
          [Symbol(kBufferCb)]: null,
          [Symbol(kBufferGen)]: null,
          [Symbol(kCapture)]: false,
          [Symbol(kBytesRead)]: 212,
          [Symbol(kBytesWritten)]: 130,
          [Symbol(connect-options)]: [Object],
          [Symbol(RequestTimeout)]: undefined
        },
        _header: 'GET /tvl HTTP/1.1\r\n' +
          'Accept: application/json, text/plain, */*\r\n' +
          'User-Agent: axios/0.26.1\r\n' +
          'Host: api.beluga.fi\r\n' +
          'Connection: close\r\n' +
          '\r\n',
        _keepAliveTimeout: 0,
        _onPendingData: [Function: nop],
        agent: Agent {
          _events: [Object: null prototype],
          _eventsCount: 2,
          _maxListeners: undefined,
          defaultPort: 443,
          protocol: 'https:',
          options: [Object: null prototype],
          requests: [Object: null prototype] {},
          sockets: [Object: null prototype],
          freeSockets: [Object: null prototype] {},
          keepAliveMsecs: 1000,
          keepAlive: false,
          maxSockets: Infinity,
          maxFreeSockets: 256,
          scheduling: 'lifo',
          maxTotalSockets: Infinity,
          totalSocketCount: 1,
          maxCachedSessions: 100,
          _sessionCache: [Object],
          [Symbol(kCapture)]: false
        },
        socketPath: undefined,
        method: 'GET',
        maxHeaderSize: undefined,
        insecureHTTPParser: undefined,
        path: '/tvl',
        _ended: true,
        res: IncomingMessage {
          _readableState: [ReadableState],
          _events: [Object: null prototype],
          _eventsCount: 4,
          _maxListeners: undefined,
          socket: [TLSSocket],
          httpVersionMajor: 1,
          httpVersionMinor: 0,
          httpVersion: '1.0',
          complete: true,
          rawHeaders: [Array],
          rawTrailers: [],
          aborted: false,
          upgrade: false,
          url: '',
          method: null,
          statusCode: 503,
          statusMessage: 'Service Unavailable',
          client: [TLSSocket],
          _consuming: true,
          _dumped: false,
          req: [Circular *1],
          responseUrl: 'https://api.beluga.fi/tvl',
          redirects: [],
          [Symbol(kCapture)]: false,
          [Symbol(kHeaders)]: [Object],
          [Symbol(kHeadersCount)]: 6,
          [Symbol(kTrailers)]: null,
          [Symbol(kTrailersCount)]: 0,
          [Symbol(RequestTimeout)]: undefined
        },
        aborted: false,
        timeoutCb: null,
        upgradeOrConnect: false,
        parser: null,
        maxHeadersCount: null,
        reusedSocket: false,
        host: 'api.beluga.fi',
        protocol: 'https:',
        _redirectable: Writable {
          _writableState: [WritableState],
          _events: [Object: null prototype],
          _eventsCount: 3,
          _maxListeners: undefined,
          _options: [Object],
          _ended: true,
          _ending: true,
          _redirectCount: 0,
          _redirects: [],
          _requestBodyLength: 0,
          _requestBodyBuffers: [],
          _onNativeResponse: [Function (anonymous)],
          _currentRequest: [Circular *1],
          _currentUrl: 'https://api.beluga.fi/tvl',
          [Symbol(kCapture)]: false
        },
        [Symbol(kCapture)]: false,
        [Symbol(kNeedDrain)]: false,
        [Symbol(corked)]: 0,
        [Symbol(kOutHeaders)]: [Object: null prototype] {
          accept: [Array],
          'user-agent': [Array],
          host: [Array]
        }
      },
      response: {
        status: 503,
        statusText: 'Service Unavailable',
        headers: {
          'cache-control': 'no-cache',
          connection: 'close',
          'content-type': 'text/html'
        },
        config: {
          transitional: [Object],
          adapter: [Function: httpAdapter],
          transformRequest: [Array],
          transformResponse: [Array],
          timeout: 0,
          xsrfCookieName: 'XSRF-TOKEN',
          xsrfHeaderName: 'X-XSRF-TOKEN',
          maxContentLength: -1,
          maxBodyLength: -1,
          validateStatus: [Function: validateStatus],
          headers: [Object],
          method: 'get',
          url: 'https://api.beluga.fi/tvl',
          data: undefined
        },
        request: <ref *1> ClientRequest {
          _events: [Object: null prototype],
          _eventsCount: 7,
          _maxListeners: undefined,
          outputData: [],
          outputSize: 0,
          writable: true,
          destroyed: false,
          _last: true,
          chunkedEncoding: false,
          shouldKeepAlive: false,
          maxRequestsOnConnectionReached: false,
          _defaultKeepAlive: true,
          useChunkedEncodingByDefault: false,
          sendDate: false,
          _removedConnection: false,
          _removedContLen: false,
          _removedTE: false,
          _contentLength: 0,
          _hasBody: true,
          _trailer: '',
          finished: true,
          _headerSent: true,
          _closed: false,
          socket: [TLSSocket],
          _header: 'GET /tvl HTTP/1.1\r\n' +
            'Accept: application/json, text/plain, */*\r\n' +
            'User-Agent: axios/0.26.1\r\n' +
            'Host: api.beluga.fi\r\n' +
            'Connection: close\r\n' +
            '\r\n',
          _keepAliveTimeout: 0,
          _onPendingData: [Function: nop],
          agent: [Agent],
          socketPath: undefined,
          method: 'GET',
          maxHeaderSize: undefined,
          insecureHTTPParser: undefined,
          path: '/tvl',
          _ended: true,
          res: [IncomingMessage],
          aborted: false,
          timeoutCb: null,
          upgradeOrConnect: false,
          parser: null,
          maxHeadersCount: null,
          reusedSocket: false,
          host: 'api.beluga.fi',
          protocol: 'https:',
          _redirectable: [Writable],
          [Symbol(kCapture)]: false,
          [Symbol(kNeedDrain)]: false,
          [Symbol(corked)]: 0,
          [Symbol(kOutHeaders)]: [Object: null prototype]
        },
        data: '<html><body><h1>503 Service Unavailable</h1>\n' +
          'No server is available to handle this request.\n' +
          '</body></html>\n'
      },
      isAxiosError: true,
      toJSON: [Function: toJSON]
    }

File: projects/beluga/index.js [hunk 1]
```diff
@@ -2,9 +2,9 @@ const utils = require('../helper/utils');
 
 async function fetch() {
   let data = await utils.fetchURL('https://api.beluga.fi/tvl')
-  return data.data.tvl;
+  return {polygon: data.data.Polygon, fantom: data.data.Fantom};
```
Comment (g1nt0ki) [inline projects/beluga/index.js:5]: the api is down?

Comment (chainvisions) [reply projects/beluga/index.js:5]: Should be up now. My apologies!

**Comment by llamatester** [comment]
Error while running adapter at projects/beluga: 
        
 
  
    
    
    Error: TVL for key tvl is not a number, instead it is NaN
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:78:11)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
        at async Promise.all (index 1)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:3

File: projects/beluga/index.js [hunk 2]
```diff
@@ -2,9 +2,9 @@ const utils = require('../helper/utils');
...
 }
 
 module.exports = {
   fetch
```
Comment (g1nt0ki) [inline projects/beluga/index.js:9]: this fetch no longe works, can you update it to:
    const utils = require('../helper/utils');

    function fetch(chain) {
      return async () => utils.fetchURL('https://api.beluga.fi/tvl').then(d => ({ 'usd-coin':  d.data[chain] }))
    }

    module.exports = {
      timetravel: false,
      polygon: {
        tvl: fetch('Polygon')
      },
      fantom: {
        tvl: fetch('Fantom')
      },
    }

Comment (chainvisions) [reply projects/beluga/index.js]: Done

**Comment by llamatester** [comment]
Error while running adapter at projects/beluga: 
        
 
  
    
    
    ReferenceError: utils is not defined
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/beluga/index.js:2:66
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:57:31)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:149:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:137:29
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:132:44
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by llamatester** [comment]
The adapter at projects/beluga exports TVL: 
        
 
 
    fantom                    2.16 M
    polygon                   22.70 k
    
    total                    2.18 M

**Pull request merged**